src/assets/scripts/airport/Arrival/ArrivalBase.js

```diff
@@ -61,150 +213,205 @@ export default class ArrivalBase {
-            if (_has(this.airport.stars, pieces[i])) {
-                star = pieces[i];
-
-                if (i > 0) {
-                    entry = pieces[i - 1];
-                }
+        // find last fix along STAR that is outside of airspace, ie: next fix is within airspace
+        // distance between closest fix outside airspace and airspace border in nm
+        let extra = 0;
+        let totalDistance = 0;
+        const waypointModelList = this.airport.findWaypointModelsForStar(
+            this.activeRouteModel.base,
+            this.activeRouteModel.origin,
+            this.airport.runway
+        );
+
+        for (let i = 0; i < waypointModelList.length; i++) {
+            const waypoint = waypointModelList[i];
+            const waypointPosition = waypoint.position;
+            let previousWaypoint = waypoint;
+            let previousPosition = waypoint.position;
+
+            if (i > 0) {
+                previousWaypoint = waypointModelList[i - 1];
+                previousPosition = previousWaypoint.position;
             }
-        }
 
-        // Find the last fix that's outside the airspace boundary
-        const fixes = this.airport.getSTAR(star, entry, runway);
-        // FIXME: this is never used. is it needed?
-        // const lastFix = fixes[0][0];
-        // distance between closest fix outside a/s and a/s border, nm
-        let extra = 0;
+            if (isWithinAirspace(this.airport, waypointPosition) && i > 0) {
+                extra = nm(calculateDistanceToBoundary(this.airport, previousPosition));
 
-        for (const i in fixes) {
-            const fix = fixes[i][0];
-            const pos = this.airport.fixes[fix].position;
-            const fix_prev = (i > 0)
-                ? fixes[i - 1][0]
-                : fix;
-            const pos_prev = (i > 0)
-                ? this.airport.fixes[fix_prev].position
-                : pos;
-
-            if (inAirspace(this.airport, pos)) {
-                if (i >= 1) {
-                    extra = nm(dist_to_boundary(this.airport, pos_prev));
-                    break;
-                }
-            } else {
-                // calculate distance between fixes
-                fixes[i][2] = nm(distance2d(pos_prev, pos));
+                continue;
             }
+
+            totalDistance += waypoint.distanceFromPreviousWaypoint;
         }
 
-        // Determine spawn offsets
-        const spawn_offsets = [];
-        // distance between succ. arrivals, nm
-        const entrail_dist = this.speed / this.frequency;
-        // TODO: replace with _map
-        const dist_total = array_sum($.map(fixes, (v) => {
-            return v[2];
-        })) + extra;
-
-        for (let i = entrail_dist; i < dist_total; i += entrail_dist) {
-            spawn_offsets.push(i);
+        // FIXME: incluing this causes aircraft to spawn within airspace. something goofy is going on here.
+        // totalDistance += extra;
```

Comment:
hmm. I'm going to have to look at this a little more.  Thanks for the info here, it certainly helps clarify things a little bit!